main/src/addins/MonoDevelop.AspNetCore/MonoDevelop.AspNetCore.Commands/PublishToFolderBaseCommandHandler.cs

```diff
@@ -0,0 +1,158 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using System.IO;
+using MonoDevelop.Components.Commands;
+using MonoDevelop.Core;
+using MonoDevelop.Core.Execution;
+using MonoDevelop.Ide;
+using MonoDevelop.Ide.Gui;
+using MonoDevelop.Projects;
+using MonoDevelop.AspNetCore.Dialogs;
+using MonoDevelop.DotNetCore;
+using MonoDevelop.Core.ProgressMonitoring;
+
+namespace MonoDevelop.AspNetCore.Commands
+{
+	abstract class PublishToFolderBaseCommandHandler : CommandHandler
+	{
+		PublishToFolderDialog dialog;
+		OutputProgressMonitor consoleMonitor;
+		protected DotNetProject project;
+
+		protected override async void Run (object dataItem)
+		{
+			try {
+				if (!(dataItem is PublishCommandItem publishCommandItem)) {
+					dialog = new PublishToFolderDialog (new PublishCommandItem (project, null));
+					dialog.PublishToFolderRequested += Dialog_PublishToFolderRequested;
+					if (dialog.Run () == Xwt.Command.Close) {
```

Comment:
```suggestion
					WindowFrame parent = Toolkit.CurrentEngine.WrapWindow (IdeApp.Workbench.RootWindow);
					if (dialog.Run (parent) == Xwt.Command.Close) {
```

Because we need it to be modal.